Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -795,18 +846,19 @@ internal struct Symbol: BSONValue {
     public var bsonType: BSONType { return .symbol }
 
     public func encode(to storage: DocumentStorage, forKey key: String) throws {
-        throw MongoError.bsonEncodeError(message: "Symbols are deprecated; use a string instead")
+        throw RuntimeError.internalError(message: "Symbols are deprecated; use a string instead")
     }
 
     public static func from(iterator iter: DocumentIterator) throws -> Symbol {
-        throw MongoError.bsonDecodeError(message:
-            "`Symbol`s are deprecated; use `Symbol.asString` to parse as a string instead")
+        throw RuntimeError.internalError(
+                message: "`Symbol`s are deprecated; use `Symbol.asString` to parse as a string instead"
+        )
     }
 
     internal static func asString(from iter: DocumentIterator) throws -> String {
         var length: UInt32 = 0
         guard let strValue = bson_iter_symbol(&iter.iter, &length) else {
-            throw MongoError.bsonDecodeError(message: retrieveErrorMsg(type: "Symbol", key: iter.currentKey))
+            throw wrongIterTypeError(iter, expected: Symbol.self)
```

Comment:
does `bson_iter_symbol` ever actually return `nil`? I guess only when it is called when the iter is not on a symbol?